src/typecheck/mod.rs

```diff
@@ -249,8 +249,10 @@ where
         )?;
     }
 
-    let result = wildcard_vars_to_type(wildcard_vars, &table);
-    let lin = linearizer.complete(building, (table, names)).into_inner();
+    let result = wildcard_vars_to_type(wildcard_vars.clone(), &table);
+    let lin = linearizer
+        .complete(building, (table, names, wildcard_vars.clone()))
```

Comment:
```suggestion
        .complete(building, (table, names, result.clone()))
```

I think you want to use `result` here. `wildcard_vars` only maps wildcard ids to unification variables, not to the final, resolved type, if I remember correctly.